plugins/modules/packaging/os/copr.py

```diff
@@ -0,0 +1,477 @@
+        state = dict()
+        repo_filename = "_copr:{0}:{1}:{2}.repo".format(self.host, self.user, self.project)
+        state["repo"] = "{0}/{1}/{2}".format(self.host, self.user, self.project)
+        state["repo_filename"] = repo_filename
+        repo_filename_path = (
+            "{0}/_copr:{1}:{2}:{3}.repo".format(self.base.conf.get_reposdir, self.host, self.user, self.project)
+        )
+        if self.state == "enabled":
+            enabled = self._enable_repo(repo_filename_path)
+            state["msg"] = "enabled"
+            state["state"] = bool(enabled)
+        elif self.state == "disabled":
+            disabled = self._disable_repo(repo_filename_path)
+            state["msg"] = "disabled"
+            state["state"] = bool(disabled)
+        elif self.state == "absent":
+            removed = self._remove_repo()
+            state["msg"] = "absent"
+            state["state"] = bool(removed)
+        return state
+
+    @staticmethod
+    def _compare_repo_content(repo_filename_path, repo_content_api):
+        """Compare the contents of the stored repository with the information from the server.
+
+        Args:
+            repo_filename_path: Path to repository.
+            repo_content_api: The information about the repository from the server.
+
+        Returns:
+            True, if the information matches, False otherwise.
+        """
+        copr_repo_file = Path(repo_filename_path)
+        if not copr_repo_file.is_file():
+            return False
+        with open(repo_filename_path, "r") as file:
+            repo_content_file = file.read()
+        return repo_content_file == repo_content_api
+
+    @staticmethod
+    def chroot_conf():
+        """Obtain information about the distribution, version, and architecture of the target.
+
+        Returns:
+            Chroot info in the form of distribution-version-architecture.
+        """
+        (distribution, version, codename) = distro.linux_distribution(full_distribution_name=False)
+        base = CoprModule.get_base()
+        return "{0}-{1}-{2}".format(distribution, version, base.conf.arch)
+
+    @staticmethod
+    def _sanitize_username(user):
+        """Modify the group name.
+
+        Args:
+            user: User name.
+
+        Returns:
+            Modified user name if it is a group name with @.
+        """
+        if user[0] == "@":
+            return "group_{0}".format(user[1:])
+        return user
+
+
+def run_module():
+    """The function takes care of the functioning of the whole ansible copr module."""
+    module_args = dict(
+        host=dict(type="str", default="copr.fedorainfracloud.org"),
+        protocol=dict(type="str", default="https"),
+        name=dict(type="str", required=True),
+        state=dict(type="str", choices=["enabled", "disabled", "absent"], default="enabled"),
+        chroot=dict(type="str"),
+    )
+    module = AnsibleModule(argument_spec=module_args, supports_check_mode=True)
```

Comment:
I don't see `module.check_mode` used anywhere in the code.

Reply:
if the check_mode actually is not supported (doesn't check anything and just returns passed parameters), this should be, at least, mentioned in the `notes:` section (though IMO would be good to implement real check_mode if possible).

Reply:
ah, i see now, there is some work inside `run`, etc.

Reply:
agreed with https://github.com/ansible-collections/community.general/pull/1592#pullrequestreview-564504363